apps/core/scenes/search/SearchModal.js

```diff
@@ -47,42 +48,42 @@ class SearchModal extends React.Component<Props> {
     this.onClose();
   };
 
+  getDefaultPlace = () => {
+    const { modalType, travelFrom, travelTo } = this.props;
+    switch (modalType) {
+      case MODAL_TYPE.ORIGIN:
+        return travelFrom;
+      case MODAL_TYPE.DESTINATION:
+        return travelTo;
+      default:
+        return '';
+    }
+  };
+
   render() {
+    const { adults, bags, infants, modalType, tripType } = this.props;
+
     return (
-      <Modal
-        visible={this.props.modalType !== MODAL_TYPE.HIDDEN}
-        onClose={this.onClose}
-      >
-        {this.props.modalType === MODAL_TYPE.TRIP_TYPE && (
+      <Modal visible={modalType !== MODAL_TYPE.HIDDEN} onClose={this.onClose}>
+        {modalType === MODAL_TYPE.TRIP_TYPE && (
           <Select
             optionsData={TRIP_TYPE}
-            selectedType={this.props.tripType}
+            selectedType={tripType}
             onSelect={this.handleTripTypeSelect}
             onClosePress={this.onClose}
           />
         )}
-        {this.props.modalType === MODAL_TYPE.PASSENGERS && (
+        {modalType === MODAL_TYPE.PASSENGERS && (
           <PassengersInputs
-            adults={this.props.adults}
-            infants={this.props.infants}
-            bags={this.props.bags}
+            adults={adults}
+            infants={infants}
+            bags={bags}
             onClosePress={this.onClose}
             onSavePress={this.handlePassengersSave}
           />
         )}
-        {[MODAL_TYPE.ORIGIN, MODAL_TYPE.DESTINATION].includes(
-          this.props.modalType,
-        ) && (
-          <View
-            style={{
-              margin: 25,
-              flexDirection: 'row',
-              alignItems: 'center',
-            }}
-          >
-            <Icon name="information-circle" />
-            <Text> TODO: Add place picker for the {this.props.modalType} </Text>
-          </View>
+        {[MODAL_TYPE.ORIGIN, MODAL_TYPE.DESTINATION].includes(modalType) && (
+          <PlacePicker type={modalType} defaultValue={this.getDefaultPlace()} />
```

Comment:
Instead of propdrilling this through PlacePicker that will not use it, could you rather connect to the context in `PlacePickerContent`

Reply:
Yes definitely currently I am working, it will be also in my next PR.
In my next PR the user can pick-up a place from suggested places.
Also, I connect the `PlacePickerContent` directly to the context.
And the PlacePicker folder will be moved to `apps/core/components` 

Is it these goals OK?